.github/workflows/windows_test.yml

```diff
@@ -30,22 +30,24 @@ jobs:
           key: viennals-dependency-cache-${{ runner.os }}-${{env.BUILD_TYPE}}-${{ hashFiles( './external/upstream/**CMakeLists.txt' ) }}
```

Comment:
https://github.com/cpm-cmake/CPM.cmake?tab=readme-ov-file#cpm_source_cache

We should use the CPM_SOURCE_CACHE in the workflows to cache